syntax = "proto3";

package flowpipe.stages.csv.source.v1;

// Configuration for CsvSource stage
message CsvSourceConfig {
  // File path to read
  string path = 1;

  // CSV delimiter (default: ",")
  string delimiter = 2;

  // Indicates if first non-empty row is a header row
  bool has_header = 3;

  // Trim whitespace around fields
  bool trim_whitespace = 4;

  // Skip empty lines
  bool skip_empty_lines = 5;
}
